include/silo/config/preprocessing_config.h

```diff
@@ -7,59 +7,52 @@
 #include <string_view>
 #include <unordered_map>
 
+#include <Poco/Util/OptionSet.h>
 #include <fmt/format.h>
 
-#include "silo/config/util/abstract_config_source.h"
+#include "config/config_interface.h"
+#include "config/source/yaml_file.h"
+#include "silo/config/config_defaults.h"
 
 namespace silo::config {
 
-const AbstractConfigSource::Option INPUT_DIRECTORY_OPTION{{"inputDirectory"}};
-const AbstractConfigSource::Option OUTPUT_DIRECTORY_OPTION = {{"outputDirectory"}};
-const AbstractConfigSource::Option INTERMEDIATE_RESULTS_DIRECTORY_OPTION = {
-   {"intermediateResultsDirectory"}
-};
-const AbstractConfigSource::Option PREPROCESSING_DATABASE_LOCATION_OPTION = {
-   {"preprocessingDatabaseLocation"}
-};
-const AbstractConfigSource::Option DUCKDB_MEMORY_LIMIT_OPTION = {{"duckdbMemoryLimitInG"}};
-const AbstractConfigSource::Option LINEAGE_DEFINITIONS_FILENAME_OPTION = {
-   {"lineageDefinitionsFilename"}
-};
-const AbstractConfigSource::Option NDJSON_INPUT_FILENAME_OPTION = {{"ndjsonInputFilename"}};
-const AbstractConfigSource::Option REFERENCE_GENOME_FILENAME_OPTION = {{"referenceGenomeFilename"}};
-
-const std::string DEFAULT_OUTPUT_DIRECTORY = "./output/";
-
 class PreprocessingConfig {
    friend class fmt::formatter<silo::config::PreprocessingConfig>;
 
+   std::optional<std::filesystem::path> default_preprocessing_config;
+   std::optional<std::filesystem::path> preprocessing_config;
+
   public:
-   std::filesystem::path input_directory = "./";
-   std::filesystem::path output_directory = DEFAULT_OUTPUT_DIRECTORY;
-   std::filesystem::path intermediate_results_directory = "./temp/";
+   std::filesystem::path input_directory;
+   std::filesystem::path output_directory;
+   std::filesystem::path intermediate_results_directory;
    std::optional<std::filesystem::path> preprocessing_database_location;
    std::optional<uint32_t> duckdb_memory_limit_in_g;
    std::optional<std::filesystem::path> lineage_definitions_file;
    std::optional<std::filesystem::path> ndjson_input_filename;
```

Comment:
It can only be a non-optional if it has a hard-coded default value via getConfigSpecification(). 

But maybe validate() should check that it is set? @Taepper

Reply:
(Not having config fields be std::optional when they are in fact required (via validate) could be a reason for a design that first collects all the values, then sets the fields in one go at the end. But it's too late to change the design now.)